### PR TITLE
feat: enables Kubernetes syncing for flagd

### PR DESCRIPTION
**Pull request by AlexsJones**
- This now will default to the Kubernetes sync provider
- It will check and append the current SA to the cluster rolebinding for the flagD notifier API

### CODE REVIEW

**Comment by beeme1mr** [comment]
Hey @AlexsJones, sorry this hasn't been reviewed yet. I've added @james-milligan and @skyerus to see if they can take a look tomorrow.

File: config/samples/end-to-end.yaml [hunk 1]
```diff
@@ -47,7 +49,38 @@ spec:
...
+---
+apiVersion: rbac.authorization.k8s.io/v1
+# This role binding allows "dave" to read secrets in the "development" namespace.
+# You need to already have a ClusterRole named "secret-reader".
+kind: ClusterRoleBinding
+metadata:
+  name: open-feature-demo-sa-role-rb
+subjects:
+- kind: ServiceAccount
+  name: open-feature-demo-sa #
```
Comment (skyerus) [inline config/samples/end-to-end.yaml:77]:
```suggestion
  name: open-feature-demo-sa
```

**Review by skyerus — APPROVED** [review]

File: config/rbac/flagd_kubernetes_sync_clusterrole.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+# Prepends to open-feature-operator-flagd-kubernetes-sync
+  name: flagd-kubernetes-sync
+rules:
+- apiGroups: ["core.openfeature.dev"] # "" indicates the core API group
```
Comment (toddbaert) [inline config/rbac/flagd_kubernetes_sync_clusterrole.yaml:7]: `# "" indicates the core API group`

I don't understand this comment.

**Review by toddbaert — APPROVED** [review]
This is a great change.

> This now will default to the Kubernetes sync provider

I think this is totally justified now.

Is it worth mentioning this automatic role-binding in the readme?

File: config/rbac/flagd_kubernetes_sync_clusterrole.yaml (hunk 2 above)
Comment (AlexsJones) [reply config/rbac/flagd_kubernetes_sync_clusterrole.yaml:7]: It's something I should remove `- apiGroups: ""` would indicate core API group, will update

**Comment by AlexsJones** [comment]
> This is a great change.
> 
> > This now will default to the Kubernetes sync provider
> 
> I think this is totally justified now.
> 
> Is it worth mentioning this automatic role-binding in the readme?

I don't think it's necessary, it's part of the machinery. But happy welcome other opinions

**Pull request merged**